pulpcore/cli/core/group.py

```diff
@@ -162,3 +155,58 @@ def remove_user(pulp_ctx: PulpContext, entity_ctx: PulpGroupUserContext, usernam
     user_pk = user_href.split("/")[-2]
     group_user_href = f"{entity_ctx.group_ctx.pulp_href}users/{user_pk}/"
     entity_ctx.delete(group_user_href)
+
+
+@group.group()
+@pass_entity_context
+@pass_pulp_context
+@click.pass_context
+def role(ctx: click.Context, pulp_ctx: PulpContext, group_ctx: PulpGroupContext) -> None:
+    pulp_ctx.needs_plugin(PluginRequirement("core", min="3.17.dev"))
+    ctx.obj = PulpGroupRoleContext(pulp_ctx, group_ctx)
+
+
+role.add_command(
+    list_command(
+        decorators=[
+            group_option,
+            click.option("--role"),
+            click.option("--role-in", "role__in"),
+            click.option("--role-contains", "role__contains"),
+            click.option("--role-icontains", "role__icontains"),
+            click.option("--role-startswith", "role__startswith"),
+            click.option("--object", "content_object", callback=null_callback),
+        ]
+    )
+)
+role.add_command(
+    create_command(
+        decorators=[
+            group_option,
+            click.option("--role", required=True),
+            click.option("--object", "content_object", required=True),
+        ]
+    ),
+    name="assign",
```

Comment:
We should agree on a consistent name for these actions. Brian was suggesting add/remove for the roles rest mixin pr.

Reply:
Also, I see this and the remove commands both require `--object` to be specified. Are we not allowing users to add/remove model level roles?

Reply:
We do allow that by specifying `--object ""` explicitely. I thought it was too dangerous to assign model level roles whenever the admin just forgot to provide this opttion.